### ISSUE TITLE
Filter, order and set limit for videos

### ISSUE THREAD

**Issue by SameenAhnaf** [TeamNewPipe/NewPipe#6145]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

Video filtering is a really useful feature. Unfortunately, Newpipe doesn't have it anywhere. Sometimes, only certain number of videos uploaded on a certain time are required. In such cases, set limit option is useful.

Following options can be added for video filtering anywhere. These options should be available as radio buttons.

1  Upload date: Last hour, today, before or after certain date
2  By video title
3. By channel name (This option might be removed from channel feed)
4. By video length
5. Features: Live, 4K, HD, Cc, creative commons, vr180, 360, 3D, HDR, Location
6. Most popular

All filtering options should come with ascending and descending order. 

Setting limits for videos will be really helpful especially when you only need a certain number of videos. "Set Limit" option should also follow "Sort by" and "Order" options.

Limit options may be like

1. Unlimited: All videos will be loaded.
2. Video number: A certain number of videos will be loaded.
3. Video length: Only videos of certain length will be loaded.
4.bTotal video length: Videos will be loading until it exceeds a certain time.
5..Keywords: Include (Single tap) or exclude (long press) videos with certain keywords in title and description. 
6. Channel: Include or exclude videos from certain channels. Not for channel  #feed.
7. Playlist: Include or exclude videos from certain playlists. Not for playlist feed.

These options could be placed under "Background, Play All and popup" row.

Default settings might be "Sort by: Date added", "Order: Descending", "Set Limit: Unlimited". However, all these settings should be seperately customisable for

1. Channel feed
2. Subscription feed
3. Playlist feed
4. Search results

#### How will you/everyone benefit from this feature?

Get your desired videos directly without wasting data.

**Comment by Angelk90** [comment]
@SameenAhnaf : There are news regarding this, in the app I do not find the similar function.


![image](https://user-images.githubusercontent.com/20476002/128170650-9d107d28-847b-4623-ac41-64b98c481c01.png)

**Comment by SameenAhnaf** [comment]
@Angelk90 Yes, there are only basic sorting for now. Advanced options are still unavailable.

![IMG_20210804_181925](https://user-images.githubusercontent.com/79245641/128179565-9f8097c2-0549-4ac1-abfd-2223bfaa0085.jpg)

**Comment by Angelk90** [comment]
@SameenAhnaf : In fact I was referring to a more complex order, for example:
sort by: upload date
